### PR TITLE
Add images to /codelabs

### LINKED ISSUE #663: Add screenshots to the codelabs page
https://webdev.dartlang.org/codelabs should have screenshots, both for esthetic reasons and because it'll help people find a codelab that corresponds to an example they've seen before.

### PR DESCRIPTION
**Pull request by kwalrath**
TODO (before committing): ~optimize and~ perhaps reduce the images. (Chrome produces prettier scaled images than the tools I've tried, so far, and maybe it's a good thing to provide a bigger version of each image...)

This will be staged at https://kw-webdev-dartlang-2.firebaseapp.com/codelabs, but for some reason it's taking forever to update. (If you go to that page and it has big red boxes, then it hasn't updated yet.)

/cc @Sfshaza 

Fixes #663

### CODE REVIEW

**Comment by kwalrath** [comment]
Btw, the CSS is based on https://github.com/dart-lang/site-www/blob/master/src/_assets/stylesheets/_books.scss.

File: src/_assets/stylesheets/_overwrites.scss [hunk 1]
```diff
@@ -132,3 +132,31 @@ img.centered {
...
+    .title {
+      margin: 0;
+    }
+    .credits {
+      margin: 0;
+      margin-bottom: 5px;
+    }
+  }
+}                                                                    
+
```
Comment (chalin) [inline src/_assets/stylesheets/_overwrites.scss]: The `_overwrites.scss` file is meant for entries that overwrite styles defined elsewhere, so these styles shouldn't be here. I would suggest copying over https://github.com/dart-lang/site-www/blob/master/src/_assets/stylesheets/_books.scss to the site-webdev `_assets/stylesheets` folder and adding an `@import '_books'` to `_assets/stylesheets/main.scss`.

File: src/_codelabs/index.md [hunk 2]
```diff
@@ -6,25 +6,111 @@ permalink: /codelabs
...
+</div>
 
-[Low-Level HTML Pirate Codelab](/codelabs/darrrt)
-: Learn how to use Dart's low-level HTML package while
-building a pirate badge generator that runs in any modern browser.
 
 For other available labs, see the
-[codelabs]({{site.dartlang}}/codelabs) page on dartlang.
+[codelabs]({{site.dartlang}}/codelabs) page on
+[www.dartlang.org.](https://www.dartlang.org)
```
Comment (chalin) [inline src/_codelabs/index.md]: Possibly drop the "www." from the link text.

File: src/_codelabs/index.md [hunk 3]
```diff
@@ -6,25 +6,111 @@ permalink: /codelabs
...
+  <div class="details">
+    <h4 class="title">
+    <a href="https://codelabs.developers.google.com/codelabs/from-java-to-dart/">Intro to Dart for Java Developers</a>
+    </h4>
+    <p>
+      Use DartPad (no download required!) to explore how
+      Dart makes writing modern apps easy and fun.
+    </p>
+  </div>
+</div>
```
Comment (chalin) [inline src/_codelabs/index.md]: Oh, that's a lot of HTML to manage. With a little tweak of the CSS we should be able to write something like the following for each of the entries in this page:
```markdown
<div class="item-with-pic">
  <img src="/codelabs/images/from-java-to-dart.png"
       width="150px"
       alt="Bicycle image from codelab">
  <div class="details" markdown="1">
  #### [Intro to Dart for Java Developers](https://codelabs.developers.google.com/codelabs/from-java-to-dart/)

  Use DartPad (no download required!) to explore how
  Dart makes writing modern apps easy and fun.
  </div>
</div>
```

Comment (chalin) [inline src/_codelabs/index.md]: (Maybe w/o the indentation.)

**Comment by kwalrath** [comment]
I've updated the files, and you can see the new version here: https://kw-webdev-dartlang-1.firebaseapp.com/codelabs

If you compare side-by-side with the first version I posted, https://kw-webdev-dartlang-2.firebaseapp.com/codelabs, you can see that there's more space between items. I'm OK with that.

**Comment by kwalrath** [comment]
I originally styled h3 & h4 directly (h3 being what the www book page uses), but that seemed broken, since I might want another style.

**Comment by chalin** [comment]
@kwalrath - Given that we have such a beautiful page now, should we consider adding back "Codelabs" to the webdev sidenav under resources?

**Comment by kwalrath** [comment]
I'll file an issue about adding back "Codelabs" to the sidenav.

**Pull request merged**